src/app/core/utils/ComponentOrganizers.ts

```diff
@@ -0,0 +1,55 @@
+import {Vector} from "Vector";
+import {Graph} from "math/Graph";
+
+import {CreateGraph, IOObjectSet} from "./ComponentUtils";
+
+import {Component} from "core/models/Component";
+
+function OrganizeCore(groups: IOObjectSet, start: Vector, depthMap: Map<number, number>): void {
+    const depthArray: Component[][] = [];
+    const components = groups.getComponents();
+
+    // Convert the map to a 2d array where index of first array is the depth
+    for (const [index, depth] of depthMap.entries()) {
+        while (depthArray.length <= depth)
+            depthArray.push([]);
+        depthArray[depth].push(components[index]);
+    }
+
+    const xOffset: number = 250;
+    const yOffset: number = 150;
```

Comment:
They are arbitrary values used to set the spacing between components. I pretty much just assigned them some random values until I found what I thought looked good enough.
The start vector that is passed in is what is used to define the top left corner where the circuit placement starts. This value in ExprToCircuitPopup is defined as `info.camera.getPos().sub(info.camera.getCenter().scale(info.camera.getZoom()/1.5));` which is intended to be near the top left corner of the camera's view, since the circuit placement expands down and to the right.

As for the second part, I don't seem to be having that issue. If I move to a different part of the scene and even if I change the zoom level, the circuit still starts to be placed in the top left corner of the camera's view.